lib/puppet/util/log.rb

```diff
@@ -146,8 +146,12 @@ def Log.newdestination(dest)
     rescue => detail
       Puppet.log_exception(detail)
 
-      # If this was our only destination, then add the console back in.
-      newdestination(:console) if @destinations.empty? and (dest != :console and dest != "console")
+      # If this was our only destination, then add the console back in,
+      # but also re-raise because we could not set up logging correctly.
+      if destinations.empty? and (dest != :console and dest != "console")
+        newdestination(:console)
+        raise detail
+      end
```

Comment:
If a valid logdest is given first, and an invalid logdest is second, then the second logdest is silently ignored, eg

```
$ bx puppet agent -td --logdest syslog --logdest /no/such/file/foo.log
Exiting; failed to retrieve certificate and waitforcert is disabled
```

If the args are reversed then it raises as expected:

```
$ bx puppet agent -td --logdest /no/such/file/foo.log --logdest syslog
Error: Permission denied @ dir_s_mkdir - /no
Error: Could not set logdest to /no/such/file/foo.log.
Error: Could not parse application options: Could not set logdest to /no/such/file/foo.log.
Permission denied @ dir_s_mkdir - /no
```

I'm thinking it would be better to always raise, not just when destinations is empty.